src/manager.ts

```diff
@@ -371,6 +371,42 @@ export class Manager extends Disposable {
 		}
 	}
 
+	/**
+	 * Runs task for workspace from within extension. Must have at least one workspace open.
+	 * @param file optional file to use to find the workspace to run the task out of.
+	 * @returns
+	 */
+	public async runTaskForFile(file?: vscode.Uri): Promise<void> {
+
+		if (!file) {
+			file = vscode.window.activeTextEditor?.document.uri;
+		}
+
+		let workspace;
+		if (file) {
+			workspace = vscode.workspace.getWorkspaceFolder(file);
+		} else if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders?.length > 0) {
+			if (this._serverGroupings.size > 0) {
+				this._serverGroupings.forEach((grouping) => {
+					// get the first available open server
```

Comment:
This is last available, not first, since your forEach loop will always run through all items. You could use `.find` if it matters